docs/rebase.md

```diff
@@ -0,0 +1,171 @@
+
+If you do not have necessary access rights ask for that branch to be created for you.
+
+### 3. Cherry-picks
+
+For this step to be easily manageable it is required to run `hack/sync-forks.sh`
+script mentioned in the prerequisites. This script will sync all `UPSTREAM` commits
+to the current kubernetes version. This in turn helps to get the list of needed
+upstream cherry picks you need to apply. There are 3 possible commit t ypes:
+
+- `carry` - we need this, otherwise the entire world will collapse
+- `12345` - specific PR number indicating that we picked something that was introduced
+  in newer version. You need to double check if the current version of kubernetes
+  you are working on has the patch. If not - make sure to carry it over.
+- `drop` - these are the trickiest ones, because they require extra check, you
+  need to ensure if the change introduced with such a commit is no longer needed.
+  Personally, I always double check with commit author about those type of changes.
+
+Once you are done with cherry picking you need to ensure at least that:
+
+1. Kubernetes compiles, iow. `make all` works.
+2. All unit tests are working - `make test`.
+3. All integration are working - `make test-integration`.
+
+Before moving on, the last commit should contain all the necessary generated
+changes, run `make update` and commit the result as a `drop` commit.
+
+Once all that is done, create a new pull request against [our kubernetes fork](https://github.com/openshift/kubernetes/),
+targeting the branch created in the previous step.
+
+### 4. Origin restore deps
+
+At this point you are ready to start the actual rebase process. Inside origin
+repository you need to run `hack/godep-restore.sh` to get the current level of
+all dependencies origin requires. It is crucial to use this script instead of
+manually invoking `godep save`, because this script add necessary forks information
+for the deps where we carry our own specific patches.
+
+In case of any problems during this step make sure to carefully examine the error
+and if needed update the script or fix the contents of `Godeps/Godeps.json` so
+that this step finishes cleanly.
+
+### 5. Checkout kubernetes
+
+After previous step go into kubernetes repository and checkout the desired version,
+you prepared in step 3 and merged into [our kubernetes fork](https://github.com/openshift/kubernetes/).
+
+### 6. Kubernetes restore deps
+
+While inside kubernetes repository run `hack/godep-restore.sh` to get the new level
+of required dependencies coming with newer version.
+
+### 7. Update deps
+
+Now that you have successfully restored origin dependencies and checked out the
+new level of kubernetes and updated its new dependencies you are now ready to
+update origin deps. To do so, go back to origin repository and run `hack/godep-save.sh`.
+This will remove entire contents of `vendor/` and `Godeps` directories and create
+new from the dependencies tree you have created in previous step.
```

Comment:
Missing the "get crazy with godep" step and those 50 modified/added lines to hack/godep-save.sh here.